tracing/src/main/java/com/palantir/tracing/DeferredTracer.java

```diff
@@ -49,26 +66,48 @@ public DeferredTracer(String operation) {
         this(Optional.of(operation));
     }
 
-    DeferredTracer(Optional<String> operation) {
-        this.trace = Tracer.copyTrace();
-        this.operation = operation;
+    /**
+     * Create a new deferred tracer, optionally specifying an operation.
+     * If no operation is specified, will attempt to use the parent span's operation name.
+     */
+    public DeferredTracer(Optional<String> operation) {
+        Optional<Trace> maybeTrace = Tracer.copyTrace();
+        if (maybeTrace.isPresent()) {
+            Trace trace = maybeTrace.get();
+            this.traceId = trace.getTraceId();
+            this.isObservable = trace.isObservable();
+            this.parentSpanId = trace.top().map(OpenSpan::getSpanId).orElse(null);
+            this.operation = operation.orElse(trace.top().map(OpenSpan::getOperation).orElse(DEFAULT_OPERATION));
```

Comment:
we could make a break and require an operation name?

I'm not sure how this is very different to what was done before?

Reply:
Previously the deferred tracers would not create a new span if the caller did not provide an operation. This is still valid because the existing trace should already have a span.